pkg/deploy/chart_manager.go

```diff
@@ -6,26 +6,28 @@ import (
 	"fmt"
 	"net"
 	"strconv"
-	"time"
+
+	moduletypes "github.com/kyma-project/module-manager/operator/pkg/types"
+
+	corev1 "k8s.io/api/core/v1"
 
 	"github.com/kyma-project/lifecycle-manager/api/v1alpha1"
 	"github.com/kyma-project/lifecycle-manager/pkg/remote"
 	modulelib "github.com/kyma-project/module-manager/operator/pkg/manifest"
-	corev1 "k8s.io/api/core/v1"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	k8syaml "sigs.k8s.io/yaml"
 )
 
+const caCertificateSecretKey = "ca.crt"
+
 var ErrExpectedExactlyOneSKRConfig = errors.New("expected exactly one SKR config")
 
 type SKRWebhookChartManager interface {
-	InstallWebhookChart(ctx context.Context, kyma *v1alpha1.Kyma,
-		remoteClientCache *remote.ClientCache, kcpClient client.Client) (bool, error)
-	RemoveWebhookChart(ctx context.Context, kyma *v1alpha1.Kyma,
-		syncCtx *remote.KymaSynchronizationContext) error
+	InstallWebhookChart(ctx context.Context, syncContext *remote.KymaSynchronizationContext) (bool, error)
```

Comment:
I see that InstallWebhookChart is always called both during Processing and Ready States. I some webhook resource is missing during `Ready` State, it force applies. Could you write a test for this use case? I have my doubts it works..

Reply:
Looking at the comment above: Force apply should never be a regular case imo and we should only use 3-way merge. Is that happening / possible right now?